tests/unit/test_xcelium_interface.py

```diff
@@ -0,0 +1,1150 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2014-2021, Lars Asplund lars.anders.asplund@gmail.com
+
+# pylint: disable=too-many-public-methods, too-many-lines
+
+"""
+Test the Xcelium interface
+"""
+
+
+import unittest
+from pathlib import Path
+import os
+from shutil import rmtree
+from unittest import mock
+from vunit.sim_if.xcelium import XceliumInterface
+from vunit.project import Project
+from vunit.ostools import renew_path, write_file, read_file
+from vunit.test.bench import Configuration
+from vunit.vhdl_standard import VHDL
+
+
+class TestXceliumInterface(unittest.TestCase):
+    """
+    Test the Xcelium interface
+    """
+
+    @mock.patch("vunit.sim_if.xcelium.XceliumInterface.find_cds_root_virtuoso")
+    @mock.patch("vunit.sim_if.xcelium.XceliumInterface.find_cds_root_xrun")
+    @mock.patch("vunit.sim_if.check_output", autospec=True, return_value="")
+    def test_compile_project_vhdl_2008(
+        self, check_output, find_cds_root_xrun, find_cds_root_virtuoso
+    ):
+        find_cds_root_xrun.return_value = "cds_root_xrun"
+        find_cds_root_virtuoso.return_value = None
+        simif = XceliumInterface(prefix="prefix", output_path=self.output_path)
+        project = Project()
+        project.add_library("lib", "lib_path")
+        write_file("file.vhd", "")
+        project.add_source_file(
+            "file.vhd", "lib", file_type="vhdl", vhdl_standard=VHDL.standard("2008")
+        )
+        simif.compile_project(project)
+        args_file = str(Path(self.output_path) / "xrun_compile_vhdl_file_lib.args")
+        check_output.assert_called_once_with(
+            [str(Path("prefix") / "xrun"), "-f", args_file], env=simif.get_env()
+        )
+        self.assertEqual(
+            read_file(args_file).splitlines(),
+            [
+                "-compile",
+                "-nocopyright",
+                "-licqueue",
+                "-nowarn DLCPTH",
+                "-nowarn DLCVAR",
+                "-v200x -extv200x -inc_v200x_pkg",
+                "-work work",
+                '-cdslib "%s"' % str(Path(self.output_path) / "cds.lib"),
```

Comment:
I strongly dislike mixing quote types for no reason. Use f-string or format to clean this up.

Reply:
I'm not sure, but it could be that the simulator really wants to see doublequotes here.

Reply:
I mentioned black earlier (I won't reference it more after this), but their docs/impl might have their str opinions about this case. Not sure if the PEPs have a recommendation for this. It doesn't look like VUnit enabled preview features so that's why it's probably not touching strings.